### PR TITLE
boards: Remove default kscan touch enabling

### PR DESCRIPTION
**Pull request by faxe1008**
This is a follow up to the PR: [pull/60867](https://github.com/zephyrproject-rtos/zephyr/pull/60867)

The goal is basically to remove the depedency of lvgl on the kscan nodes. This means in any case the `defconfigs` where `KSCAN` and `LV_Z_POINTER_KSCAN` are enabled need to be edited. 

I split the removal of the kscan nodes into a separate commit since I am not sure if the fact that this will break downstream applications breaks any sort of deprecation timeline. If its fine I can squash the commit 3 into the first one.

If having the pointer be a freestanding node is a concern, I'd be happy to change `zephyr,lvgl-pointer-input` to be similar to the way `kscan-input` worked prior. Feedback on this would also be appreciated :^)

### CODE REVIEW

File: boards/arm/bl5340_dvk/Kconfig.defconfig [hunk 1]
```diff
@@ -104,15 +104,12 @@ config BT_HCI_VS
 # https://www.buydisplay.com/2-8-inch-tft-touch-shield-for-arduino-w-capacitive-touch-screen-module
 if DISPLAY
 
-if KSCAN
-
 config INPUT_FT5336
 	default y
```
Comment (fabiobaltieri) [inline boards/arm/bl5340_dvk/Kconfig.defconfig:108]: this one is redundant, it's already default enabled, you may drop it as well while you are at it

File: boards/posix/native_posix/native_posix.dts [hunk 2]
```diff
@@ -43,6 +43,11 @@
 		};
 	};
 
+	lvgl_pointer {
```
Comment (fabiobaltieri) [inline boards/posix/native_posix/native_posix.dts:44]: ah, this changes the ordinals for native posix so you have to update https://github.com/zephyrproject-rtos/zephyr/blob/main/tests/misc/check_init_priorities/validate_check_init_priorities_output.py#L10-L17 with the output you see in https://github.com/zephyrproject-rtos/zephyr/pull/62107/checks?check_run_id=16353533961

Comment (faxe1008) [reply boards/posix/native_posix/native_posix.dts:44]: Ahh, I was a bit confused why the test broke with seemingly unrelated errors :)

Comment (fabiobaltieri) [reply boards/posix/native_posix/native_posix.dts:44]: yeah, may have to make the test a bit "smarter" if this become a hassle, though native_posix does not get changed often so maybe it's fine

**Review by fabiobaltieri — DISMISSED** [review]

File: boards/shields/g1120b0mipi/g1120b0mipi.overlay [hunk 3]
```diff
@@ -20,11 +15,17 @@
...
+		compatible = "zephyr,lvgl-pointer-input";
+		input = <&ft3267>;
+		invert-y;
+	};
 };
 
 &nxp_mipi_i2c {
 	status = "okay";
-	ft3267@38 {
+	ft3267: ft3267@38 {
```
Comment (erwango) [inline boards/shields/g1120b0mipi/g1120b0mipi.overlay:28]: I don't know how these nodes have been spared by the renaming, but they now should be called :
ft3267_<shield_name>.
See https://github.com/zephyrproject-rtos/zephyr/pull/50526.

Not directly related so not blocking.
But would be nice to fix this while you're at

Comment (faxe1008) [reply boards/shields/g1120b0mipi/g1120b0mipi.overlay:28]: Since I touched the files, might as well do this one too :^)

**Comment by faxe1008** [comment]
Addressed @erwango note regarding naming of nodes within shields.
@fabiobaltieri  Should I be squashing commit 3 into commit 1?

**Comment by fabiobaltieri** [comment]
> Should I be squashing commit 3 into commit 1?

Up to you, does it break the build to have that change in its own commit?

**Comment by fabiobaltieri** [comment]
> Build should succeed regardless of the squashing. If you want to I'll squash, for me its fine as is :^).

Ok so fine as is.

**Review by fabiobaltieri — APPROVED** [review]

**Review by erwango — APPROVED** [review]
Approved (even if I can't see the proposed changed, so will have to be fixed in an upcoming PR)

**Pull request merged**